blob/azureblob/azureblob.go

```diff
@@ -223,6 +226,18 @@ type AccountKey string
 // https://docs.microsoft.com/en-us/azure/storage/common/storage-dotnet-shared-access-signature-part-1
 type SASToken string
 
+// CloudEnvironment is an Azure Cloud Environment to target (i.e. AzureCloud, AzureUSGovernment, AzureChinaCloud).
+// It is read from the AZURE_CLOUD_ENVIRONMENT environment variable.
+type CloudEnvironment string
+
+// Mappings from the AZURE_CLOUD_ENVIRONMENT value to their respective blob storage domains.
+var cloudEnvMap = map[CloudEnvironment]string{
```

Comment:
Is there an advantage to having this mapping? I.e., why not have the CloudEnvironment variable just be the storage domain itself?

For example, with the mapping approach we'll need to update the code every time there's a new environment.

Reply:
The purpose of the mapping is to be able to use the same environment variable values ("AzureCloud", "AzureUSGovernment", etc.) that users may already be familiar with for this functionality (especially if they use Ansible, which is fairly popular). Furthermore, the values supplied mirror the Azure CLI and Azure Powershell names for these Cloud environments (when you do an "az cloud list" for example). It seemed a little more user-friendly in the end. 

However, if you think this setup is too hard to maintain into the future (I'm unsure how often Azure modifies or adds environments/endpoints), I can easily change the option to simply take in a specific URL suffix instead, it's certainly more flexible. The environment variable name should probably also be changed though (maybe something like GOCLOUD_AZURE_ENVIRONMENT?), so as not to collide with Ansible users who will be using AZURE_CLOUD_ENVIRONMENT with different values. Would you like me to do this instead?